### PR TITLE
[Enhancement] Improve English Translations

### PR DESCRIPTION
**Pull request by FreshLlamanade**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
All other items on the checklist are not applicable.

### Motivation and Context
I have been using Artemis since my first day at TUM two years ago. While I greatly appreciate the functionality of the platform, I think the overall quality of the English translations is worthy of improvement in order to provide a more professional user experience. As an English native speaker, I feel particularly capable of contributing to this part of Artemis. And so I decided to take the first step toward effecting the change I wanted to see.

### Description
I went through all 126 .json files in the directory `src\main\webapp\i18n\en` and found 90 with potential for improvement. I tried to be as conservative as possible with my changes; I was not aiming for a complete overhaul. However, in some cases I went so far as to reword entire sentences if I felt it necessary. Every commit description contains an explanation of all changes made. I apologize for the generic commit titles; by the time I realized, I was 30 commits in.

### Steps for Testing
Use Artemis in the English language mode. Essentially all of Artemis's nooks and corners have text in them.

### Test Coverage
Tests should behave precisely as before; there is no need for additional testing.

### CODE REVIEW

**Comment by krusche** [comment]
Wow, thanks for this contribution, which is highly appreciated. As soon as I find some time, I will review the changes and check if they are consistent with our understanding

File: src/main/webapp/i18n/en/assessmentInstructions.json
Comment (krusche) [inline src/main/webapp/i18n/en/assessmentInstructions.json:15]: We call it `Example Solution` on purpose to emphasize that it is one exemplary solution, but not the only solution.

`Sample solution` translates to `Musterlösung` in German, which we would like to avoid, because some users might associate this term with other solutions automatically being wrong

File: src/main/webapp/i18n/en/exampleSubmission.json
Comment (krusche) [inline src/main/webapp/i18n/en/exampleSubmission.json:12]: here, I would also prefer if we can keep the word `example submission`

Comment (krusche) [reply src/main/webapp/i18n/en/exampleSubmission.json:12]: also in the cases below

File: src/main/webapp/i18n/en/exercise.json
Comment (krusche) [inline src/main/webapp/i18n/en/exercise.json:45]: Example Solution

File: src/main/webapp/i18n/en/assessmentInstructions.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/assessmentInstructions.json:15]: Ah, understood! I will revert these changes then.

**Review by rs-maker — DISMISSED** [review]
Wow reading all of this took a long time i cant imagine how much time you put into making all these changes. Very cool and i think a great improvement to our sometimes very german english.

File: src/main/webapp/i18n/en/complaintResponse.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/complaintResponse.json:16]: Yeah, I saw that. I would go so far as to say the German version should be changed too; I think tutors would be overall less prone to pressing the wrong button while evaluating complaints, which is something I saw happens a lot (or more than it should). Would be curious to hear what others have to say about this change.

File: src/main/webapp/i18n/en/student-dashboard.json
Comment (joschahenningsen) [inline src/main/webapp/i18n/en/student-dashboard.json:20]: We should use either _register_ or _sign up_ consistently.

File: src/main/webapp/i18n/en/statistics.json
Comment (joschahenningsen) [inline src/main/webapp/i18n/en/statistics.json:19]: This could stay as is, otherwise I'd rename the variable.

File: src/main/webapp/i18n/en/result.json
Comment (joschahenningsen) [inline src/main/webapp/i18n/en/result.json:31]: Variable name is inconsistent with it's content.

**Review by joschahenningsen — APPROVED** [review]
This is a great addition, thank you so much for taking the time! 
Just a few thoughts.

File: src/main/webapp/i18n/en/student-dashboard.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/student-dashboard.json:20]: I agree, consistency could be improved upon. In that specific context I think both are viable options; "register" would be more formal but "sign up" is somehow friendlier. "Enroll" is another option we might want to consider. Does anyone else have any thoughts? I'll keep considering this in the meantime.

File: src/main/webapp/i18n/en/statistics.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/statistics.json:19]: I disagree purely on the basis that "participation" is not quantifiable; it's a singular-only noun. I think the variable should be renamed, but that was outside the scope of this PR.

File: src/main/webapp/i18n/en/result.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/result.json:31]: This is the case with quite a lot of changes; I felt it was in the interest of keeping the PR cohesive to ignore that at first and then fix it later. Would be willing to author that subsequent PR.

**Comment by FreshLlamanade** [comment]
I did not dismiss any stale reviews... how odd.

**Review by joschahenningsen — APPROVED** [review]
re-approve 

> I did not dismiss any stale reviews

They are automatically dismissed when you push changes :)

File: src/main/webapp/i18n/en/complaintResponse.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/complaintResponse.json:16]: I'd like to propose leaving this change as-is.

File: src/main/webapp/i18n/en/error.json
Comment (ndricimrr) [inline src/main/webapp/i18n/en/error.json:21]:
```suggestion
        "feedbackDetailTextNull": "Error! If feedback is present, it must contain feedback detail text.",
```
something along those lines, as **all elements** is not valid here. There is only one element in context, as a feedback has a single detailed text field.

File: src/main/webapp/i18n/en/error.json
Comment (ndricimrr) [inline src/main/webapp/i18n/en/error.json:18]:
```suggestion
        "submissionMissing": "Error! The submission is not connected with the result anymore. Your changes were rejected.",
```

File: src/main/webapp/i18n/en/course.json
Comment (ndricimrr) [inline src/main/webapp/i18n/en/course.json:107]:
```suggestion
                "description": "Activate this checkbox to define the number of presentations a student has to hold to receive the grade bonus for the course. Afterwards you can define for every created exercise whether this feature should be activated."
```

**Review by ndricimrr — COMMENTED** [review]
Looks good 👍 . Only some minor suggestions

File: src/main/webapp/i18n/en/error.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/error.json:21]: Thanks for pointing that out!

**Review by argertboja — COMMENTED** [review]
Everything looks good! Nice job 👍🏼 I have only added a few minor suggestions 😃

File: src/main/webapp/i18n/en/guidedTour.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/guidedTour.json:16]: Ahah! This is a matter of British English vs American English. I didn't think to change "cancelled" because that is the form I am used to as an American. The Artemis team may have a policy of using BRE, and that's fine. I noticed that "modeling" is also spelled with only one L, which is distinctly British.

File: src/main/webapp/i18n/en/guidedTourProgrammingExercise.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/guidedTourProgrammingExercise.json:15]: Good change, I should have caught that.

File: src/main/webapp/i18n/en/guidedTourProgrammingExercise.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/guidedTourProgrammingExercise.json:34]: Sounds much better!

File: src/main/webapp/i18n/en/guidedTour.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/guidedTour.json:16]: Hold on, just Googled this. Apparently "cancelled" is British English after all and I've been using that version all my life without knowing it. In that case I would suggest leaving it as-is.

Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/guidedTour.json:16]: According to Grammarly I am dead wrong about spellings with one L being British. It comes as a big surprise to me, but that's purely an American thing! I am not used to those spellings at all. I think I will make another commit changing "modeling" to "modelling"; that is the more international spelling, after all.

**Comment by ndricimrr** [comment]
Checked again. Looks good 👍

**Review by coolchock — DISMISSED** [review]
going through all changes was a very time consuming task, but everything looks good to me :+1:

File: src/main/webapp/i18n/en/exam.json
Comment (lschlesinger) [inline src/main/webapp/i18n/en/exam.json:211]: maybe we should either use "submission state" or "Submitted status", I think it is both referring to the value of the variable `submitted` being true (for submitted) or false (for unsubmitted)

File: src/main/webapp/i18n/en/fileUploadExercise.json
Comment (lschlesinger) [inline src/main/webapp/i18n/en/fileUploadExercise.json:14]: most of the times, the value for key "typeNameToConfirm" does not use capitalized nouns, e.g. "the name of the course / lecture / team ...". Maybe we can change the occurrences of this standard sentence to always _not_ capitalize the noun.

**Review by lschlesinger — CHANGES_REQUESTED** [review]
Great improvement for Artemis!!
I had some nitpicky comments, but one thing that seems important to me is the replacement of "Participation" with "Participant", I think that might be a mistake and would appreciate if you could clarify the reason for the change.

Edit: I obviously did not submit the comment for the participation issue, so I will add it here. 
I think we indeed meant the participation (german: Teilnahme), which is associated with submissions, a result, etc. When reading "Delete Participant", I would assume, that we delete a participating person or team, but actually, we are deleting the object "Participation" with the associated submission, result, etc. and do _not_ change the participant(s). Is the word Participation wrong here, I saw that the related commit message indicated a "misuse of the word participation" but, after my explanation, is it really a misuse?

File: src/main/webapp/i18n/en/activate.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/activate.json:5]: I saw that and left it as-is because it seemed intentional (maybe another string gets appended to it?). Would like clarification as well.

File: src/main/webapp/i18n/en/apollonDiagram.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/apollonDiagram.json:36]: I am not quite sure what you mean. Are you suggesting we change "Apollon Diagram" in line 34 to "Quiz Exercise"?

Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/apollonDiagram.json:36]: Nevermind, I see what you meant. Yes I agree.

File: src/main/webapp/i18n/en/assessment.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/assessment.json:19]: See above

File: src/main/webapp/i18n/en/complaint.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/complaint.json:51]: See above. If it turns out that every instance of a period missing from the end of a sentence is a mistake, then I'll gladly add them in. Otherwise I have no idea which strings should and should not have a period at the end.

File: src/main/webapp/i18n/en/course.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/course.json:37]: Capitalization, like the periods at the ends of sentences, is another aspect which is heavily context-dependent and that I purposefully left as-is in most cases (I can think of one commit where I changed capitalization quite heavily, but alas). In this case I think the inconsistent capitalization is actually fine; line 36 uses "questions and answers" as improper nouns, whereas line 37 refers to the "Questions and Answers section" as a proper noun.

File: src/main/webapp/i18n/en/fileUploadExercise.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/fileUploadExercise.json:14]: I agree with you, and there are lots of other places where I would have liked to remove capitalization as well. There were so many instances of this though, with a lot of them falling into sort of gray areas where you could argue either way about it being a proper or improper noun that I figured I would deal with capitalization another time. I'll make a commit for this though; it's clear that the noun is improper and should be lowercase.

File: src/main/webapp/i18n/en/exam.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/exam.json:211]: Could you make a concrete change suggestion so I can better see what you mean?

File: src/main/webapp/i18n/en/apollonDiagram.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/apollonDiagram.json:31]: Looking at this case again in context I find it unlikely that it would cause any problems. Making the change

File: src/main/webapp/i18n/en/exam.json
Comment (lschlesinger) [reply src/main/webapp/i18n/en/exam.json:211]: sorry, that was a bit vague :) here the suggestions:
```
"submitText": "This will set the submission date of the exam to this moment, and the submission state to 'submitted'. Un-submitting the student exam later is possible.\n",
"unSubmitText": "This will delete the submission date and set the submission state to 'unsubmitted'.",
```
Would you say "set to this moment" should be replaced by "set to this point in time"? I just thought about that.

File: src/main/webapp/i18n/en/complaint.json
Comment (lschlesinger) [reply src/main/webapp/i18n/en/complaint.json:51]: I also don't want to put all the burden on you of changing to more consistent punctuation and capitalization. As this PR is about translation, we should maybe focus on the language itself. @krusche Do you see the value in putting all that effort into revising punctuation and capitalization now?

File: src/main/webapp/i18n/en/participation.json
Comment (lschlesinger) [inline src/main/webapp/i18n/en/participation.json:3]: > I think we indeed meant the participation (german: Teilnahme), which is associated with submissions, a result, etc. When reading "Delete Participant", I would assume, that we delete a participating person or team, but actually, we are deleting the object "Participation" with the associated submission, result, etc. and do not change the participant(s). Is the word Participation wrong here, I saw that the related commit message indicated a "misuse of the word participation" but, after my explanation, is it really a misuse?

@FreshLlamanade what about the general comment I left? Should we reconfirm the use of "Participation" here?

File: src/main/webapp/i18n/en/activate.json
Comment (lschlesinger) [reply src/main/webapp/i18n/en/activate.json:5]: Ah well, I see, it's indeed appended to: "Please sign in" with sign in being an anchor tag! So no need for changes here 👍

File: src/main/webapp/i18n/en/exam.json
Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/exam.json:211]: Those changes look good to me!
And I have no opinion one way or the other on those two options, I think they're equally good/clear

File: src/main/webapp/i18n/en/guidedTour.json
Comment (krusche) [reply src/main/webapp/i18n/en/guidedTour.json:16]: in any case, we would like to consistently use American English and **`not`** British English

File: src/main/webapp/i18n/en/complaint.json
Comment (krusche) [reply src/main/webapp/i18n/en/complaint.json:51]: I think we have more important issues than punctuation and capitalization, but if there are volunteers, I also would not object against it :)

File: src/main/webapp/i18n/en/participation.json
Comment (krusche) [reply src/main/webapp/i18n/en/participation.json:3]: I would also argue that participation (Teilnahme) is correct. At least this is what DeepL tells me as appropriate translation

Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/participation.json:3]: Ah, I forgot to respond to that. I agree with what you wrote there and acknowledge that "participant" does not always capture what was meant by "Teilnahme". The trouble is, there is simply no good way to use the word "participation" in that way. I have to hold firm that it is not quantifiable; you can't have "a" participation or multiple "participations" in English. I am trying to be as polite as possible when I say that it made me cringe the first time I saw it used that way.
That said, in cases like "Delete Participant" I find myself doubting whether the change was really necessary in the first place. "Delete Participation" is fine; I think I got ahead of myself and failed to see that it was not actually quantified there. I will go through these changes and revert what I can.

Comment (lschlesinger) [reply src/main/webapp/i18n/en/participation.json:3]: >  I am trying to be as polite as possible when I say that it made me cringe the first time I saw it used that way. 😆 

Maybe we could just avoid the plural and only use `Participation`. 
Imagine we replace the marked occurrences with `Participation`, IMO it would be ok, even if an overview of several teams/students or individual participation (e.g. more than more than one) is displayed.
Bildschirmfoto 2021-05-27 um 21 10 32

Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/participation.json:3]: I think that would surely put us on a better track than before. "8 Participation" sounds like more of a mistake than just using the incorrect plural, though. In that case I really think "8 Participants" communicates the same information.

Comment (lschlesinger) [reply src/main/webapp/i18n/en/participation.json:3]: ah true, I did not see the number. We could also just not show the number here (the total sum of entries in this list is shown at the bottom anyways). Or in this case use participants, indeed. I'm afraid that has to be decided on a case-by-case basis.

Comment (krusche) [reply src/main/webapp/i18n/en/participation.json:3]: Please keep the number 😅

Comment (lschlesinger) [reply src/main/webapp/i18n/en/participation.json:3]: Feels like I've opened Pandora's box 😄

Comment (krusche) [reply src/main/webapp/i18n/en/participation.json:3]: DeepL tells me there is a plural **participations**

I also found:
https://simple.wiktionary.org/wiki/participations

Maybe it's just not a typical term used in some English speaking countries?

Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/participation.json:3]: I don't know what else to say than DeepL is incorrect 😅 I guarantee that participation is quite strongly uncountable. It sounds absurd to me and my American peers

Comment (FreshLlamanade) [reply src/main/webapp/i18n/en/participation.json:3]: I've pushed a commit to try and find a middle ground in a few cases

File: src/main/webapp/i18n/en/participation.json
Comment (lschlesinger) [inline src/main/webapp/i18n/en/participation.json:11]: I am not sure, if with identifier {{ param }} would be better for updated and deleted as well... But if I am correct, these translations are currently not used anyways.
All other changes that were made with 2dd4bce0c85fbd93b3abf1d782d679d05b89e150 seem good to me 👍

**Review by MaisiKoleni — CHANGES_REQUESTED** [review]
I have looked through most of the changes, and they look good to me.

However, I have one major concern: we have to be cautious with "participations" because many of the occurrences cannot be replaced with "participants". The participant is the person/user, and while the table for "Participations" also lists the users and thus the participants by name, you actually see their participation, which in Artemis is a concept referring to a concrete application (and solution domain) domain object, the `Participation`.

We should discuss how we want to handle this in the future. "I disagree purely on the basis that "participation" is not quantifiable" seems to apply to the majority of use cases, but a quick search shows there is disagreement about whether the plural "participations" exists.

Most classic dictionaries don't list "participations", so that would be an argument to replace it. Some uses in business, law and politics suggests that it is unusual and perhaps odd but accurate in the particular context. This is also the case in Artemis, so I am not convinced whether we should simply replace it with "participants". We should get more opinions on this.

Edit: just saw there are already comments on this. So, we already have a discussion on that.

**Comment by FreshLlamanade** [comment]
@MaisiKoleni Your points make a lot of sense. I think each individual instance will require a fair amount of consideration as to what is the ideal solution. Sometimes we want to refer to the concrete application domain object (e.g. the set of a student's submissions), in which case I suppose the plural "Participations" is perfectly valid to use, but sometimes we instead use participation in a more general sense - the students' _involvement_ in the exercise, exam, etc.; in those cases I would rather stray away from quantifying the word unnecessarily. I will look over all changes again and think about this.

**Comment by lschlesinger** [comment]
@FreshLlamanade, Is there anything that needs to be clarified other than the question of participation? I have proposed this PR for the test session later today and we will specifically check whether the use of `participation` and `participant` is appropriate in the different contexts. In my opinion, it is time for this contribution to finally become visible 💪

**Comment by FreshLlamanade** [comment]
@lschlesinger I think the participation issue is the last real sticking point. I've pushed one last commit with two small changes related to that. Glad to know that we are on the home stretch!

**Comment by lschlesinger** [comment]
@FreshLlamanade could you merge develop and solve the conflict please? I am sorry to ask you to, but this is required for merging 😄 and also, since our bamboo instance was not too stable during the last days, the build was not triggered and I cannot deploy your branch. I hope the build will be successful after merging develop and push again. If so, I will deploy and test the branch.

**Comment by FreshLlamanade** [comment]
@lschlesinger Gladly!

**Comment by FreshLlamanade** [comment]
I am actually not seeing any of the merge conflicts locally... not sure how to fix them. Never run into this before. Does anyone know what might be the problem?

**Comment by lschlesinger** [comment]
> I am actually not seeing any of the merge conflicts locally... not sure how to fix them. Never run into this before. Does anyone know what might be the problem?

Is there any other way to contact you (besides github comments), e.g. slack? Then I would help you with merging develop and solving conflicts. If you are using intelliJ, you would see the conflicts in the git section (where you would also track your local changes) after merging.

**Comment by FreshLlamanade** [comment]
@lschlesinger I appreciate it. I have fixed merge conflicts before, it's simply that they are not showing up locally for me. I googled and found [this thread](https://github.community/t/merge-conflicts-on-github-not-locally/949/6) that describes what is happening; it seems I would need to be able to resolve it directly on GitHub which I cannot do without write access... someone else will have to do it in that case.

**Comment by lschlesinger** [comment]
@FreshLlamanade I think this is due to the fact, that you forked the repository. I made a PR on your branch to merge 99 commits into `FreshLlamanade:develop` from `ls1intum:develop`. I hope, with this procedure, you are able to solve the conflicts.

**Comment by FreshLlamanade** [comment]
Okay, all merge conflicts are fixed. I made sure to follow along with the changes made in other commits, so all the string additions and name changes have been carried over, and in some cases I spruced up the content of the new strings as well.

**Comment by FreshLlamanade** [comment]
Next PR I am going to remember to create a new branch; both branches being named `develop` has been a bit of a headache.

**Pull request merged**
